framework/src/base/Assembly.C

```diff
@@ -365,6 +365,28 @@ Assembly::buildLowerDFE(FEType type) const
   }
 }
 
+void
+Assembly::buildLowerDDualFE(FEType type) const
+{
+  if (!_fe_shape_data_dual_lower[type])
+    _fe_shape_data_dual_lower[type] = new FEShapeData;
```

Comment:
This needs to be freed in `Assembly::~Assembly()`.

Reply:
nailed it

Reply:
Can't we rather use smart pointers?

Reply:
nice catch!